mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/work/impl/PojoWorkPlanImpl.java

```diff
@@ -39,6 +39,8 @@
 	private final Map<Class<?>, PojoIndexedTypeWorkPlan<?, ?, ?>> indexedTypeDelegates = new LinkedHashMap<>();
 	private final Map<Class<?>, PojoContainedTypeWorkPlan<?>> containedTypeDelegates = new HashMap<>();
 
+	private volatile boolean isPreparing = false;
```

Comment:
This class is not thread safe and should only be used in a single thread. You can remove the `volatile` keyword.